### ISSUE TITLE
Autobahn WebSocket error when called from Electron main process

### ISSUE THREAD

**Issue by jt-helsinki** [crossbario/autobahn-js#459]
When trying to connect to a socket server from Electron's main process the Autobahn code tries to use the native WebSocket implementation. It needs to use the `ws` implementation. 

The error I receive is:

> could not create WAMP transport 'websocket':  browser does not support WebSocket or WebSocket in Web workerr

On line 102 of `websocket.js` the following line:

```
if (global.process && global.process.versions.node
       && !global.process.versions.hasOwnProperty('electron') && !global.process.__nwjs) {
```
This does not allow for the `ws` WebSocket implementation to be used from the main process. If I remove `!global.process.versions.hasOwnProperty('electron')` then Connecting works without an error.

As a solution, would it be an idea to allow the user to optionally specify the WebSocket implementation when creating a connection?

**Comment by oberstet** [comment]
The automatic detection of the WebSocket implementation to use - for all scenarios AutobahnJS is able to run under (in principle) - is indeed notoriously brittle. There are a couple of issues on the bug tracker here touching on this. Having a documented/official (optional/additional) way of _manually_ choosing could be a way out of this mess indeed ..

**Comment by oberstet** [comment]
maybe we should instead of positively identifying the environment (like electron), simply check if there is `global.WebSocket` already there or not?

**Comment by jt-helsinki** [comment]
Yes, for most cases the implementation can be automatically selected but where this doesn't work the programmer should be able to simply configure which implementation to use. 

I think it is best to keep it simple and just let the programmer specify which implementation to use.
